### ISSUE TITLE
Game Mode doesn't AutoMount

### ISSUE THREAD

**Issue by pcwizard7** [scawp/Steam-Deck.Mount-External-Drive#16]
After a reboot, the drive isn't being auto mounted in game mode, When I switch to desktop mode, the drive is there and mounted. And it shows up in steam library folders again

I Can Run zMount .sh in game mode and mount the partition from there.  But it should auto mount at boot for gamer mode, right?

Thanks

**Comment by Qlink84** [comment]
same problem here. can you fix this ?

**Comment by Bboy486** [comment]
Same issue for me as well. FWIW I am using a USB hub but that shouldn't matter.

**Comment by xfechx** [comment]
I am having the same issue. Wasn't the whole purpose of the script is to auto-mount when hot plugged under Game mode?

There is no need to use the script if you do not need automount outside of Game mode, you can do this directly on settings under removable media. And it will auto mount under desktop mode.

**Comment by pcwizard7** [comment]
> I am having the same issue. Wasn't the whole purpose of the script is to auto-mount when hot plugged under Game mode?
> 
> There is no need to use the script if you do not need automount outside of Game mode, you can do this directly on settings under removable media. And it will auto-mount under desktop mode.

I switch to this method this tool adds btrfs support but I only added support and not convert the OS partition btw as that is an option 
https://gitlab.com/popsulfr/steamos-btrfs

This adds btrfs for windows
https://github.com/maharmstone/btrfs

once you got them installed you need to take all content off your sd card then use the game mode format sd card button in settings which will format it btrfs  then move all contents back on and then make sure sd card is added to steam again and your done

**Comment by Bboy486** [comment]
My workaround has been to boot to desktop then plug in the USB hub and SD card and wait for it to mount then restart into game OS.

**Comment by xfechx** [comment]
Yes, but then the script is pointless. You can do this without any script.

**Comment by Bboy486** [comment]
I am aware but let's give the dev a chance to provide a solution.

**Comment by xfechx** [comment]
Yes, that's out of question. I also consider it is important to make people
who aren't aware that at this point this is the issue and also save people
time and headeaches and also save the dev from more people rasing the same
issue.

**Comment by lazerwalker** [comment]
Also running into this, curious if folks have come up with a good solution. Considering btrfs, but it seems like it has some downsides as well.

**Comment by xfechx** [comment]
The current versions of the OS already support this, no need for external
scripts. It should auto mount your usb drive at any time you plug it.

**Comment by lazerwalker** [comment]
What I'm experiencing is my internal NTFS partition gets auto-mounted in desktop mode, but it fails to auto-add the SteamLibrary directory to my library, and trying to do so manually also fails silently.

**Comment by xfechx** [comment]
Today I struggled the whole day with a similar issue, it was related to a
particular SD card mounting on read mode, so a few times I've had to
unmount and mount again and fixes itself. Also today I had to manually
change a path from /Games/Steam/steamapps to /SteamLibrary. Not sure if
some changes are happening and breaking things..
